ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSshell.cpp

```diff
@@ -85,6 +86,17 @@ string execute_shell_for_output(const string &command) {
   PROCESS_INFORMATION pi = { };
   if (CreateProcessW(NULL, ctstr_command, NULL, NULL, TRUE, CREATE_NO_WINDOW, NULL, NULL, &si, &pi)) {
     while (WaitForSingleObject(pi.hProcess, 5) == WAIT_TIMEOUT) {
+      DWORD dwRead = 0;
+      DWORD dwAvail = 0;
+      PeekNamedPipe(hStdOutPipeRead, NULL, 0, NULL, &dwAvail, NULL);
+      if (dwAvail) {
+        vector<char> buffer(dwAvail);
+        BOOL success = ReadFile(hStdOutPipeRead, &buffer[0], dwAvail, &dwRead, NULL);
```

Comment:
Just a note, you don't have to do this, but `&buffer[0]` is also the same as `buffer.data()` which you could arguably say is cleaner.